crates/project_panel/src/project_panel_tests.rs

```diff
@@ -5821,6 +5821,103 @@ async fn test_hide_root(cx: &mut gpui::TestAppContext) {
     }
 }
 
+#[gpui::test]
+async fn test_compare_selected_files(cx: &mut gpui::TestAppContext) {
+    init_test_with_editor(cx);
+
+    let fs = FakeFs::new(cx.executor().clone());
+    fs.insert_tree(
+        "/root",
+        json!({
+            "file1.txt": "content of file1",
+            "file2.txt": "content of file2",
+            "dir1": {
+                "file3.txt": "content of file3"
```

Comment:
Great that we have 3 files!

Now we can easily extend this test to ensure selecting more files changes the diff input, same as the diff text.
We will be able to also test that we open a new diff each time we do this (by asserting that we have 2 items of the same kind now), which is nice too.